public/app/plugins/gen.go

```diff
@@ -54,52 +53,52 @@ func main() {
-	})
+	// composableKindsGen := codejen.JennyListWithNamer(func(decl *corecodegen.DeclForGen) string {
+	// 	return decl.Meta.Common().MachineName
+	// })
 
 	// All the jennies that comprise the composable kinds generator pipeline
-	composableKindsGen.Append(
-		corecodegen.GoTypesJenny("pkg/tsdb", nil),
-		corecodegen.TSTypesJenny("public/app/plugins", &corecodegen.TSTypesGeneratorConfig{
-			GenDirName: func(decl *corecodegen.DeclForGen) string {
-				// FIXME this hardcodes always generating to experimental dir. OK for now, but need generator fanout
-				return filepath.Join(decl.Meta.Common().MachineName, "x")
-			},
-		}),
-	)
-
-	composableKindsAdapter := AdaptTest[*corecodegen.DeclForGen](composableKindsGen, func(ptDecl *codegen.PluginTreeDeclForGen) []*corecodegen.DeclForGen {
-		log.Printf("adapting plugin: %s", ptDecl.Tree.RootPlugin().Meta().Id)
-		slots := ptDecl.Tree.RootPlugin().SlotImplementations()
-		decls := []*corecodegen.DeclForGen{}
-
-		for k, slot := range slots {
-			log.Printf("  slot: %s, path: %s", slot.Name(), slot.Underlying().Path().String())
-			someDecl := &kindsys.SomeDecl{
-				V: slot.Underlying(),
-				Meta: kindsys.ComposableMeta{
-					CommonMeta: kindsys.CommonMeta{
-						Name:              k,
-						MachineName:       k,
-						PluralName:        k + "s",
-						PluralMachineName: k + "s",
-					},
-					CurrentVersion: slot.Latest().Version(),
-				},
-			}
-
-			decl := corecodegen.DeclForGenFromLineage(someDecl, slot)
-			decls = append(decls, decl)
-		}
-
-		return decls
-	})
-
-	pluginKindGen.AppendManyToMany(composableKindsAdapter)
+	// composableKindsGen.Append(
+	// 	corecodegen.GoTypesJenny("pkg/tsdb", nil),
+	// 	corecodegen.TSTypesJenny("public/app/plugins", &corecodegen.TSTypesGeneratorConfig{
+	// 		GenDirName: func(decl *corecodegen.DeclForGen) string {
+	// 			// FIXME this hardcodes always generating to experimental dir. OK for now, but need generator fanout
+	// 			return filepath.Join(decl.Meta.Common().MachineName, "x")
+	// 		},
+	// 	}),
+	// )
+
+	// composableKindsAdapter := AdaptTest[*corecodegen.DeclForGen](composableKindsGen, func(ptDecl *codegen.PluginTreeDeclForGen) []*corecodegen.DeclForGen {
+	// 	log.Printf("adapting plugin: %s", ptDecl.Tree.RootPlugin().Meta().Id)
+	// 	slots := ptDecl.Tree.RootPlugin().SlotImplementations()
+	// 	decls := []*corecodegen.DeclForGen{}
+
+	// 	for k, slot := range slots {
+	// 		log.Printf("  slot: %s, path: %s", slot.Name(), slot.Underlying().Path().String())
+	// 		someDecl := &kindsys.SomeDecl{
+	// 			V: slot.Underlying(),
+	// 			Meta: kindsys.ComposableMeta{
+	// 				CommonMeta: kindsys.CommonMeta{
+	// 					Name:              k,
+	// 					MachineName:       k,
+	// 					PluralName:        k + "s",
+	// 					PluralMachineName: k + "s",
+	// 				},
+	// 				CurrentVersion: slot.Latest().Version(),
+	// 			},
+	// 		}
+
+	// 		decl := corecodegen.DeclForGenFromLineage(someDecl, slot)
+	// 		decls = append(decls, decl)
```

Comment:
@sdboyer in addition I struggled quite a lot to wrap my head around how to convert `(*codegen.PluginTreeDeclForGen).Tree.RootPlugin().SlotImplementations()` to `[]*corecodegen.DeclForGen`. I'm guessing we need to implement something we don't have today, but I was kind of already under the impression that the slot/lineage is already "thema bound" via `ParsePluginFS` and "ready to be consumed". If you think you can give some pointers that would be helpful and/or we can schedule a meeting to discuss it

Reply:
Yes, you're definitely right, these loaders need to be written. Also, existing `models.cue` have to be refactored. And that spec for what the `grafanaplugin` overall instance looks like probably needs to be written. Here's what i handwaved before when we paired -

```cue
package grafanaplugin

import "github.com/grafana/grafana/pkg/kindsys"

kindsys.#PluginDeclSpace

composable: Panel: {
	lineage: name
}
customType: Foo: {
	lineage: {
		compose: kindsys.slot.Panel
	}
}
```

Reply:
I moved the example out of a comment and made a whole standalone issue - #59002